src/silo/query_engine/filter_expressions/has_aa_mutation.cpp

```diff
@@ -0,0 +1,79 @@
+#include "silo/query_engine/filter_expressions/has_aa_mutation.h"
+
+#include <nlohmann/json.hpp>
+#include <utility>
+#include <vector>
+
+#include "silo/common/aa_symbols.h"
+#include "silo/query_engine/filter_expressions/aa_symbol_equals.h"
+#include "silo/query_engine/filter_expressions/negation.h"
+#include "silo/query_engine/filter_expressions/or.h"
+#include "silo/query_engine/operators/operator.h"
+#include "silo/query_engine/query_parse_exception.h"
+
+#include "silo/database.h"
+
+namespace silo::query_engine::filter_expressions {
+
+HasAAMutation::HasAAMutation(std::string aa_sequence_name, uint32_t position)
+    : aa_sequence_name(std::move(aa_sequence_name)),
+      position(position) {}
+
+std::string HasAAMutation::toString(const silo::Database& /*database*/) const {
+   std::string res = std::to_string(position);
+   return res;
+}
+
+std::unique_ptr<operators::Operator> HasAAMutation::compile(
+   const silo::Database& database,
+   const silo::DatabasePartition& database_partition,
+   AmbiguityMode mode
+) const {
+   const char ref_symbol =
+      database.aa_sequences.at(aa_sequence_name).reference_sequence.at(position);
+
+   if (mode == UPPER_BOUND) {
```

Comment:
What does `UPPER_BOUND` actually stand for? Can we rename it to something more descriptive?

Reply:
The old names were `EXACT` and `MAYBE`, but I liked the interpretation of upper bounds and lower bounds as well, as then the implementation is way easier to wrap the head around. (As described in Chaoran's paper, exact returns the lower bound of sequences matching a filter and maybe returns the upper bound)

Reply:
I added a doc comment to the struct declaration

Reply:
A doc comment also sounds like a good idea :+1: